provider-example/src/hpke.rs

```diff
@@ -0,0 +1,97 @@
+use std::fmt::{Debug, Formatter};
+
+use hpke_rs_crypto::types::{AeadAlgorithm, KdfAlgorithm, KemAlgorithm};
+use hpke_rs_crypto::HpkeCrypto;
+use hpke_rs_rust_crypto::HpkeRustCrypto;
+
+use rustls::crypto::hpke::{
+    EncapsulatedSecret, Hpke, HpkePrivateKey, HpkeProvider, HpkePublicKey, HpkeSuite,
+};
+use rustls::Error;
+
+pub static HPKE_PROVIDER: &'static dyn HpkeProvider = &HpkeRsHpkeProvider {};
+
+/// A Rustls HPKE provider backed by hpke-rs.
+#[derive(Debug)]
+struct HpkeRsHpkeProvider {}
```

Comment:
`HpkeRsHpkeProvider` -- I feel like `HpkeRsProvider` would be `Hpke` enough? (Same for other types with similar naming...)

Reply:
Agreed, `HpkeRsHpkeProvider` looks a bit like keyboard barf :laughing: 

Fixed up.